src/main/scala/ru/org/codingteam/horta/protocol/jabber/JabberProtocol.scala

```diff
@@ -52,26 +54,34 @@ class JabberProtocol() extends Actor with ActorLogging {
     case Reconnect(otherConnection) =>
       log.info(s"Ignored reconnect request from connection $otherConnection")
 
-    case JoinRoom(jid, locale, nickname, greeting) =>
+    case message@JoinRoom(jid, locale, nickname, greeting) =>
       log.info(s"Joining room $jid")
       val actor = context.actorOf(
         Props(new MucMessageHandler(locale, self, jid, nickname)), jid)
 
-      val muc = new MultiUserChat(connection, jid)
-      rooms = rooms.updated(jid, RoomDefinition(muc, actor))
-
-      muc.addMessageListener(new MucMessageListener(jid, actor, log))
-      muc.addParticipantStatusListener(new MucParticipantStatusListener(muc, actor))
-
-      val filter = new AndFilter(new PacketTypeFilter(classOf[Message]), new FromContainsFilter(jid))
-      connection.addPacketListener(
-        new MessageAutoRepeater(context.system, self, context.system.scheduler, jid, context.dispatcher),
-        filter)
-
-      muc.join(nickname)
-      greeting match {
-        case Some(message) => muc.sendMessage(message)
-        case None =>
+      try {
+        // TODO: Move this code to the room actor and use "let it fall" strategy ~ F
+        val muc = new MultiUserChat(connection, jid)
+        rooms = rooms.updated(jid, RoomDefinition(muc, actor))
+
+        muc.addMessageListener(new MucMessageListener(jid, actor, log))
+        muc.addParticipantStatusListener(new MucParticipantStatusListener(muc, actor))
+
+        val filter = new AndFilter(new PacketTypeFilter(classOf[Message]), new FromContainsFilter(jid))
+        connection.addPacketListener(
+          new MessageAutoRepeater(context.system, self, context.system.scheduler, jid, context.dispatcher),
+          filter)
+
+        muc.join(nickname)
+        greeting match {
+          case Some(text) => muc.sendMessage(text)
+          case None =>
+        }
+      } catch {
+        case t: Throwable =>
```

Comment:
Generally, I do not like this approach. But I cannot suggest anything better for now.

I'm ok with this if we at least will log the exception somewhere. At least its message. Like "Cannot join because of <t.message()>" or something.